### ISSUE TITLE
ignoreImports for no-unused-vars

### ISSUE THREAD

**Issue by errorx666** [eslint/eslint#9969]
**What rule do you want to change?**
`no-unused-vars`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option `ignoreImports` `boolean` <ins> or `string[]` of modules to ignore</ins> (default: `false`).

**Please provide some example code that this change will affect:**

```js
import foo, { bar } from 'baz';

let qux;
```

**What does the rule currently do for this code?**
```
'foo' is defined but never used
'bar' is defined but never used
'qux' is defined but never used
```

**What will the rule do after it's changed?**
```
'qux' is defined but never used
```

**Comment by errorx666** [comment]
Justification: In my spec tests I import a lot of utility functions.  Not all of my specs need them, but it's easier just to import everything everywhere.  This generates a lot of noise from `no-unused-vars`.  I do want to be warned if my tests declare a variable they don't need, because that probably indicates a broken test, however, the false positives from the imports make this less useful.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal. Personally, I'm not sure whether this stylistic preference is common enough to be worth adding to the core `no-unused-vars` rule -- my impression is that most users would prefer to omit the imports that aren't used. However, let's see what the rest of the team thinks. (Regardless of whether the team decides to implement this, it would also be possible for you to implement this behavior separately as part of a custom rule.)

**Comment by errorx666** [comment]
If I was to implement it and submit a PR, would it be accepted?
Is there a case against adding it?  It's opt-in, and shouldn't affect users who don't want/need it.

**Comment by platinumazure** [comment]
> If I was to implement it and submit a PR (with docs and tests), would it be accepted?

You might find this explanation useful for our [process of accepting issues](https://eslint.org/docs/maintainer-guide/issues#accepting-issues).

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
